solutions/ha-nva/layer-7-ingress/README.md

```diff
@@ -0,0 +1,106 @@
+# Highly Available Ingress with layer 7 NVAs
+
+This sample template will deploy an Azure Application Gateway in front of two simulated network virtual appliance (NVA) VMs.  The NVA VMs will forward requests on port 80 to a Web-tier Azure Load Balancer, which will distribute traffic across two back-end VMs running a simple web server.
+
+For more information on this deployment and other highly available NVA options, see [Deploy highly available NVAs](https://docs.microsoft.com/azure/architecture/reference-architectures/dmz/nva-ha).
+
+![Highly Available Ingress with layer 7 NVAs architectural diagram.](l7-ingress.png)
+
+This sample architecture leverages concepts from the [Example Parameterized Deployment With Linked Templates](https://azure.microsoft.com/resources/templates/301-parameterized-linked-templates/) and [Create ssh-keys and store in Key Vault](https://azure.microsoft.com/resources/templates/201-deployment-script-ssh-key-gen/) samples.
+
+## Deploy this template to Azure
+
+*Note: If you fork this repository, you will need to modify the link in [README.md](README.md) to point to your repo.  If you create a separate branch for testing, you will have to include a change to this link to point to your branch as well. You must include a URL-encoded link to the raw [azuredeploy.json](azuredeploy.json) file after `/uri/` in the link defined for the deployment button. You should also change the default value of `_artifactsLocation`.*
+
+#### Azure portal
+
+Use the following buttons to deploy the architecture using the Azure portal or to visualize with [armviz.io](https://armviz.io).
+
+[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmspnp%2Fsamples%2Fmaster%2Fsolutions%2Fha-nva%2Flayer-7-ingress%2Fazuredeploy.json)
+[![Deploy To Azure US Gov](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazuregov.svg?sanitize=true)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmspnp%2Fsamples%2Fmaster%2Fsolutions%2Fha-nva%2Flayer-7-ingress%2Fazuredeploy.json)
+[![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2Fmspnp%2Fsamples%2Fmaster%2Fsolutions%2Fha-nva%2Flayer-7-ingress%2Fazuredeploy.json)
+
+#### Azure CLI
+
+Use the following command to create a resource group for the deployment.
+
+```bash
```

Comment:
Consider updating code block decorator to `azurecli-interactive`.

Reply:
Does this do anything for markdown hosted from GitHub?  Or does this make things nicer in the sample browser?  I was noticing some strangeness when viewing the hub/spoke README on GitHub and I'm guessing that's because the markdown has two different publish points that we're trying to accommodate. Example: 
![image](https://user-images.githubusercontent.com/1796255/109215018-8e309580-7780-11eb-823c-0351ed42d55a.png)

Reply:
Ah, this is a good point. For code block decoration, it has no affect on presentation in GitHub, however will enable integrated shell when viewed .via our code browser.

What you are seeing above is a button that navigates from the GitHub asset back to the RA. Not ideal for sure, we can reconsider using the button controll.

Reply:
I changed the code block decorators as suggested.